reframe/frontend/dependencies.py

```diff
@@ -24,39 +25,26 @@ def build_deps(cases, default_cases=None):
     '''
 
     # Index cases for quick access
-    def build_partition_index(cases):
-        if cases is None:
-            return {}
-
-        ret = {}
-        for c in cases:
-            cname, pname = c.check.name, c.partition.fullname
-            ret.setdefault((cname, pname), [])
-            ret[cname, pname].append(c)
-
-        return ret
-
-    def build_cases_index(cases):
+    def build_index(cases):
         if cases is None:
             return {}
 
         ret = {}
         for c in cases:
             cname = c.check.name
-            pname = c.partition.fullname
-            ename = c.environ.name
-            ret.setdefault((cname, pname, ename), c)
+            ret.setdefault(cname, [])
+            ret[cname].append(c)
 
         return ret
 
-    def resolve_dep(target, from_map, fallback_map, *args):
-        errmsg = 'could not resolve dependency: %s -> %s' % (target, args)
+    def resolve_dep(target, from_map, fallback_map, key):
```

Comment:
This function signature is deceiving. The first argument is not the target, but actually the source and the `key` is the target. Also you can build the indexes just above this function and use them without passing them as an argument. This will make this function's signature simpler:

```suggestion
    def resolve_dep(src, dst):
```

I suggest using `src` and `dst` for consistency with the udeps functions.